### PR TITLE
Feature/api calls

### LINKED ISSUE #2: use apiCalls to retrieve state data
As a developer, 
When I retrieve the state data, 
I want to see the use of an apiCall,
So that I know the data displayed is updated on page load.

### PR DESCRIPTION
**Pull request by Codo-Baggins**
## What's this PR do?
- This PR fetches coronavirus API data for both the entire United States and the state of Colorado

## Where should the reviewer start?
- The reviewer should first start by looking at the APICalls.tsx file and then navigate over to the `componentDidMount()` function on App.tsx

## How should this be manually tested?
- Clone down the repo
- `cd` into the cloned down directory
- Run `npm start` 
- Navigate to the localhost URL provided in the terminal
- Open up the dev tools and open the `Components` inspection tool 
- Click on any of the data that is displayed onto the page and observe what the raw data looks like in the console

## Any background context you want to provide?
-  Styling still has not been done.  This creates some readability issues for the time being.  

## Issues Resolved 
- Fixes #2 
- Fixes #17

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -1,42 +1,119 @@
...
+class App extends Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      selectedUSAState: {}
+      allUSAData: {
+        positive: 0,
+        death: 0
+      }, 
+      selectedUSAState: {
```
Comment (alyssajoyner) [inline src/App/App.tsx:35]: We might want to reach out to a mentor to confirm if this is the best initial state data.

File: src/App/App.tsx [hunk 2]
```diff
@@ -1,42 +1,119 @@
...
 }
 
 interface State {
-  selectedUSAState: {};
+  allUSAData: Details, 
+  selectedUSAState: Details
 }
 
-class App extends Component<Props,State> {
+interface Details {
```
Comment (alyssajoyner) [inline src/App/App.tsx:18]: Will need to circle back to the use of optional parameters to ensure they are being used and passed correctly.

File: src/App/App.tsx [hunk 3]
```diff
@@ -1,42 +1,119 @@
...
+  date?: number,
+  state?: string,
+  positive: number,
+  hospitalizedCurrently?: number,
```
Comment (kncaputo) [inline src/App/App.tsx:22]: Let's touch base with Jev about the option type members

**Review by alyssajoyner — APPROVED** [review]
Nice work! Excited to get this styled on the next branch.

**Review by kncaputo — APPROVED** [review]

**Pull request merged**